### ISSUE TITLE
can not gem i -g on non existing gem

### ISSUE THREAD

**Issue by mpapis** [rubygems/rubygems#1033]
Using rubygems abf89e04954b8a2ddfb6948d0860111990a6efa2
```bash
gem -v
2.4.2
```
with `.gemspec`:
```ruby
Gem::Specification.new do |spec|
  spec.name        = 'long-nonexisting-gem-name'
  spec.version     = '1.0.0'
  spec.add_dependency('session')
end
```
with `Gemfile`:
```ruby
#ruby=2.0.0
source "https://rubygems.org"
gemspec
```
running `gem i -g` I get:
```bash
ERROR:  While executing gem ... (Gem::UnsatisfiableDependencyError)
    Unable to resolve dependency: user requested 'long-nonexisting-gem-name (= 1.0.0)'
```
and after `bundle install` (to get `Gemfile.lock`),  `gem i -g` I gives:
```bash
ERROR:  While executing gem ... (Gem::GemNotFoundException)
    unable to find ./long-nonexisting-gem-name.gemspec for gem long-nonexisting-gem-name
```
and finally after
```bash
mv .gemspec ./long-nonexisting-gem-name.gemspec
```
`gem i -g` I gives:
```ruby
ERROR:  While executing gem ... (URI::InvalidURIError)
    bad URI(is not URI?): ?gems=session (>= 0)
```

**Comment by mpapis** [comment]
and also #1204

**Comment by mpapis** [comment]
the above commit fixes the last problem described here, now against which branch should I open pull request?

**Comment by mpapis** [comment]
fixed in #1276

**Comment by aflatter** [comment]
This is still an issue.
